sqladmin/models.py

```diff
@@ -807,11 +807,13 @@ async def get_object_for_details(self, value: Any) -> Any:
 
         return await self._get_object_by_pk(stmt)
 
-    async def get_object_for_edit(self, value: Any) -> Any:
-        stmt = self._stmt_by_identifier(value)
+    async def get_object_for_edit(self, request: Request) -> Any:
+        stmt = self.edit_form_query(request)
 
-        for relation in self._form_relations:
-            stmt = stmt.options(joinedload(relation))
+        if type(self).edit_form_query == ModelView.edit_form_query:
+            # If edit_form_query fn hasn't been overridden, add all relationships
+            for relation in self._form_relations:
```

Comment:
Can we move this part to the `edit_form_query`? That way we can avoid this condition which doesn't seem nice.

Reply:
I think we can, but I'll wait for your feedback before modifying the code.